src/network/network_client.cpp

```diff
@@ -687,8 +687,17 @@ NetworkRecvStatus ClientNetworkGameSocketHandler::Receive_SERVER_ERROR(Packet *p
 
 	StringID err = STR_NETWORK_ERROR_LOSTCONNECTION;
 	if (error < (ptrdiff_t)lengthof(network_error_strings)) err = network_error_strings[error];
-
-	ShowErrorMessage(err, INVALID_STRING_ID, WL_CRITICAL);
+	if (error == NETWORK_ERROR_KICKED) {
+		char msg[255];
+		p->Recv_string(msg, sizeof(msg));
```

Comment:
how does this work when the message was not provided on the other end?

Reply:
By coincidence, there was a null byte at msg[0] when I tested my code, due to garbage on the stack.
I have added a check if there is an extra byte in the packet. If so, I assume that it is the kick message. I do this because the packet interface does not seem to have a function to check for the existence of a string.